compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -175,8 +175,10 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
       tp match
         case tp: TypeRef =>
           tp.prefix match
-            case NoPrefix if level > levelOf(tp.symbol) =>
-              tryHeal(tp.symbol, tp, pos)
+            case NoPrefix if level > levelOf(tp.symbol) && !tp.typeSymbol.hasAnnotation(defn.InternalQuoted_QuoteTypeTagAnnot) =>
+              tp.dealias match
+                case `tp` => tryHeal(tp.symbol, tp, pos)
+                case tp1 => apply(tp1)
```

Comment:
I wasn't clear in the previous comment. I mean what about doing dealiasing at line 175?

Reply:
No, that is too much dealiasing. There are a few aliases that we care about, such as the type splices.
I will revert this change.